pkg/utils/validator/resourceoverride.go

```diff
@@ -84,26 +84,47 @@ func validateResourceOverrideResourceLimit(ro fleetv1alpha1.ResourceOverride, ro
 }
 
 // validateOverridePolicy checks if override rule is selecting resource by name.
-func validateOverridePolicy(policy *fleetv1alpha1.OverridePolicy) error {
+func validateOverridePolicy(policy *placementv1alpha1.OverridePolicy) error {
 	allErr := make([]error, 0)
 	for _, rule := range policy.OverrideRules {
 		if rule.ClusterSelector == nil {
-			continue
-		} else if len(rule.ClusterSelector.ClusterSelectorTerms) == 0 {
-			allErr = append(allErr, fmt.Errorf("clusterSelector must have at least one term"))
+			allErr = append(allErr, fmt.Errorf("invalid overrideRule %v: clusterSelector is required", rule))
+		} else {
+			for _, selector := range rule.ClusterSelector.ClusterSelectorTerms {
+				// Check that only label selector is supported
+				if selector.PropertySelector != nil || selector.PropertySorter != nil {
+					allErr = append(allErr, fmt.Errorf("invalid clusterSelector %v: only labelSelector is supported", selector))
+					continue
+				}
+				if selector.LabelSelector == nil {
+					allErr = append(allErr, fmt.Errorf("invalid clusterSelector %v: labelSelector is required", selector))
+				} else if err := validateLabelSelector(selector.LabelSelector, "cluster selector"); err != nil {
+					allErr = append(allErr, err)
+				}
+			}
+		}
+		if err := validateJSONPatchOverride(rule.JSONPatchOverrides); err != nil {
+			allErr = append(allErr, err)
+		}
+	}
+	return errors.NewAggregate(allErr)
+}
+
+// validateJSONPatchOverride checks if JSON patch override is valid.
+func validateJSONPatchOverride(jsonPatchOverrides []placementv1alpha1.JSONPatchOverride) error {
+	allErr := make([]error, 0)
+	for _, patch := range jsonPatchOverrides {
+		switch {
+		case strings.Contains(patch.Path, "/kind") || strings.Contains(patch.Path, "/apiVersion"):
```

Comment:
what if the path is "/abc/kind" and "/apiVersionabc"?

we should not fail the request.